File: CS/WinSolution.Module.Win/PopupListViewController.cs
using DevExpress.ExpressApp;
using DevExpress.XtraEditors;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.BaseImpl;

namespace WinSolution.Module.Win {
    public class ShowPopupController : ViewController {
        public ShowPopupController() {
            PopupWindowShowAction showPopupAction = new PopupWindowShowAction(
                this,
                "ShowPopup",
                DevExpress.Persistent.Base.PredefinedCategory.View
            );
            showPopupAction.CustomizePopupWindowParams += this.showPopupAction_CustomizePopupWindowParams;
        }
        private void showPopupAction_CustomizePopupWindowParams(object sender, CustomizePopupWindowParamsEventArgs e) {
            e.View = Application.CreateListView(e.Application.CreateObjectSpace(), typeof(Person), false);
        }
    }
    public class ActionInPopupController : ViewController {
        public ActionInPopupController() {
            SimpleAction actionInPopup = new SimpleAction(this,
                "ActionInPopup",
                DevExpress.Persistent.Base.PredefinedCategory.PopupActions
            );
            //Dennis: Refer to the http://documentation.devexpress.com/#Xaf/CustomDocument2815 help article to see how to reorder Actions within the PopupActions container.
            actionInPopup.Execute += actionInPopup_Execute;
        }
        void actionInPopup_Execute(object sender, SimpleActionExecuteEventArgs e) {
            XtraMessageBox.Show(string.Format("Hello {0}!", e.CurrentObject));
        }
    }
}

File: VB/WinSolution.Module.Win/PopupListViewController.vb
Imports DevExpress.ExpressApp
Imports DevExpress.XtraEditors
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.BaseImpl

Namespace WinSolution.Module.Win

    Public Class ShowPopupController
        Inherits ViewController

        Public Sub New()
            Dim showPopupAction As PopupWindowShowAction = New PopupWindowShowAction(Me, "ShowPopup", DevExpress.Persistent.Base.PredefinedCategory.View)
            AddHandler showPopupAction.CustomizePopupWindowParams, AddressOf showPopupAction_CustomizePopupWindowParams
        End Sub

        Private Sub showPopupAction_CustomizePopupWindowParams(ByVal sender As Object, ByVal e As CustomizePopupWindowParamsEventArgs)
            e.View = Application.CreateListView(e.Application.CreateObjectSpace(), GetType(Person), False)
        End Sub
    End Class

    Public Class ActionInPopupController
        Inherits ViewController

        Public Sub New()
            Dim actionInPopup As SimpleAction = New SimpleAction(Me, "ActionInPopup", DevExpress.Persistent.Base.PredefinedCategory.PopupActions)
            'Dennis: Refer to the http://documentation.devexpress.com/#Xaf/CustomDocument2815 help article to see how to reorder Actions within the PopupActions container.
            AddHandler actionInPopup.Execute, AddressOf actionInPopup_Execute
        End Sub

        Private Sub actionInPopup_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs)
            XtraMessageBox.Show(String.Format("Hello {0}!", e.CurrentObject))
        End Sub
    End Class
End Namespace
